[macOS-arm64] NativeObjCMarshalTests.cpp fails to compile

```
[ 99%] Building CXX object Interop/ObjectiveC/ObjectiveCMarshalAPI/CMakeFiles/NativeObjCMarshalTests.dir/NativeObjCMarshalTests.cpp.o
/Users/stmaclea/git/runtime/src/tests/Interop/ObjectiveC/ObjectiveCMarshalAPI/NativeObjCMarshalTests.cpp:102:19: error: cast to smaller integer type 'int' from 'void *' [-Werror,-Wvoid-pointer-to-int-cast]
        int val = (int)cxt;
                  ^~~~~~~~
1 error generated.
make[2]: *** [Interop/ObjectiveC/ObjectiveCMarshalAPI/CMakeFiles/NativeObjCMarshalTests.dir/NativeObjCMarshalTests.cpp.o] Error 1
make[1]: *** [Interop/ObjectiveC/ObjectiveCMarshalAPI/CMakeFiles/NativeObjCMarshalTests.dir/all] Error 2
```

--- a/src/tests/Interop/ObjectiveC/ObjectiveCMarshalAPI/NativeObjCMarshalTests.cpp
+++ b/src/tests/Interop/ObjectiveC/ObjectiveCMarshalAPI/NativeObjCMarshalTests.cpp
@@ -99,7 +99,7 @@ namespace
     [[noreturn]]
     void ThrowInt(void* cxt)
     {
-        int val = (int)cxt;
+        int val = (int)(size_t)cxt;
         throw val;
     }
 

